### PR TITLE
Add B_All_White_Blood_Cancer_Microscopic_Image Dataset

### PR DESCRIPTION
**Pull request by yihangx**

### CODE REVIEW

**Comment by yihangx** [comment]
@googlebot I signed it!

________________________________
发件人: googlebot <notifications@github.com>
发送时间: 2019年9月25日 6:46
收件人: tensorflow/datasets <datasets@noreply.github.com>
抄送: Yihang Xin <yihang.xin@duke.edu>; Author <author@noreply.github.com>
主题: Re: [tensorflow/datasets] Add B_All_White_Blood_Cancer_Microscopic_Image Dataset (#1024)


Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

📝 Please visit https://cla.developers.google.com/<https://urldefense.proofpoint.com/v2/url?u=https-3A__cla.developers.google.com_&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=27ijxjF2Vfm1-TET89n4bWau9El2SKnzJWRPjchlkFE&e=> to sign.

Once you've signed (or fixed any issues), please reply here with @googlebot I signed it! and we'll verify it.

________________________________
What to do if you already signed the CLA
Individual signers

  *   It's possible we don't have your GitHub username or you're using a different email address on your commit. Check your existing CLA data<https://urldefense.proofpoint.com/v2/url?u=https-3A__cla.developers.google.com_clas&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=HgnsLUQD5fbBz-hIN8rcolbF8JILCopLFjsyjQy8EOs&e=> and verify that your email is set on your git commits<https://urldefense.proofpoint.com/v2/url?u=https-3A__help.github.com_articles_setting-2Dyour-2Demail-2Din-2Dgit_&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=NB5yPE44RplJKaDylfnxidbHQseFBbGDQ4ufaqkgWsw&e=>.

Corporate signers

  *   Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to go/cla#troubleshoot<https://urldefense.proofpoint.com/v2/url?u=http-3A__go_cla-23troubleshoot&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=NaQGme9aAmlT0eN5ELABvzl-EY_T7spZX4gLlQCDab0&e=> (Public version<https://urldefense.proofpoint.com/v2/url?u=https-3A__opensource.google.com_docs_cla_-23troubleshoot&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=yMLwS9jkXleYnL5gxUxNz0a_fgMGc5P4BZT72mJcJDY&e=>).
  *   The email used to register you as an authorized contributor must be the email used for the Git commit. Check your existing CLA data<https://urldefense.proofpoint.com/v2/url?u=https-3A__cla.developers.google.com_clas&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=HgnsLUQD5fbBz-hIN8rcolbF8JILCopLFjsyjQy8EOs&e=> and verify that your email is set on your git commits<https://urldefense.proofpoint.com/v2/url?u=https-3A__help.github.com_articles_setting-2Dyour-2Demail-2Din-2Dgit_&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=NB5yPE44RplJKaDylfnxidbHQseFBbGDQ4ufaqkgWsw&e=>.
  *   The email used to register you as an authorized contributor must also be attached to your GitHub account<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_settings_emails&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=PQL6IXMcMFl9oYgCl-84gBRNVS-_Mq1qkVtEG0lwXHc&e=>.

ℹ️ Googlers: Go here<https://urldefense.proofpoint.com/v2/url?u=https-3A__goto.google.com_prinfo_https-253A-252F-252Fgithub.com-252Ftensorflow-252Fdatasets-252Fpull-252F1024&d=DwMFaQ&c=imBPVzF25OnBgGmVOlcsiEgHoG1i6YHLR0Sj_gZ4adc&r=dHY9FaVfYdPqanz_CIrA_Q&m=bShMD9jTdS8keHde9X_7uNUEQALKyj_kzOrSSIELqjI&s=KxQjM8qxV-LLq84ON-FHVLNxB_0cjIVMTFU8Z95WNCw&e=> for more info.

File: tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  organization={Springer}
+}
+//@under review{
+title={GCTI-SN: Geometry-Inspired Chemical and Tissue Invariant Stain Normaliz
+ation of Microscopic Medical Images},
+author={Anubha Gupta, Rahul Duggal, Ritu Gupta, Lalit Kumar, Nisarg Thakkar,
+and Devprakash Satpathy}
+}
+
+"""
```
Comment (Ouwen) [inline tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py]: Remove blank spaces

File: tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow as tf
+import tensorflow_datasets as tfds
+
+Image_Data = ("https://drive.google.com/uc?aut"
```
Comment (Ouwen) [inline tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py]: rename to `_IMAGE_DATA`

File: tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+"""
+
+_DESCRIPTION = """\
+ A large data set of  B-ALL white blood cancer microscopic images. The dataset
+ contains total 1867 cropped images, which 1219 malignant cells images are
+ labeled as class 1, 647 normal cells images are labeled as class 0. The size
+ of each image is roughly 300 x 300 x 3 RGB.
+"""
```
Comment (Ouwen) [inline tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py]: Move `"""` to line 69

File: tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    def _info(self):
+        return tfds.core.DatasetInfo(
+            builder=self,
+            description=_DESCRIPTION,
+            features=tfds.features.FeaturesDict({
+                "image": tfds.features.Image(),
+                "label": tfds.features.ClassLabel(names=["cancer", "normal"]),
+            }),
+            supervised_keys=("image", "label"),
+            urls=["https://competitions.codalab.org/competitions/20395"],
```
Comment (Ouwen) [inline tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py]: Move urls outside of class as `_URLS`

File: tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        extracted_path = dl_manager.manual_dir
```
Comment (Ouwen) [inline tensorflow_datasets/image/B_All_White_Blood_Cancer_Microscopic_Image.py]: These collections are freely available to browse, download, and use for commercial, scientific and educational purposes as outlined in the Creative Commons Attribution 3.0 Unported License. Questions may be directed to help@cancerimagingarchive.net. Please be sure to acknowledge both this data set and TCIA in publications by including the following citations in your work:

You must have an automatic download process. No manual download.

**Comment by Ouwen** [comment]
@yuqin50

**Comment by yuqin50** [comment]
> We found a Contributor License Agreement for you (the sender of this pull request), but were 


> We found a Contributor License Agreement for you (the sender of this pull request), but were unable to find agreements for all the commit author(s) or Co-authors. If you authored these, maybe you used a different email address in the git commits than was used to sign the CLA ([login here](https://cla.developers.google.com/) to double check)? If these were authored by someone else, then they will need to sign a CLA as well, and confirm that they're okay with these being contributed to Google.
> In order to pass this check, please resolve this problem and then comment `@googlebot I fixed it.`. If the bot doesn't comment, it means it doesn't think anything has changed.
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Ftensorflow%2Fdatasets%2Fpull%2F1024) for more info**.
@googlebot I fixed it.

**Pull request closed without merging**